### PR TITLE
IGNITE-23434 Sql. Provide system view for active locks

### PR DESCRIPTION
**Pull request by xtern**
https://issues.apache.org/jira/browse/IGNITE-23434

### CODE REVIEW

File: modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/HeapLockManager.java [hunk 1]
```diff
@@ -251,22 +251,18 @@ public void releaseAll(UUID txId) {
         }
     }
 
+    @Override
+    public Iterator<Lock> locks() {
```
Comment (zstan) [inline modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/HeapLockManager.java:255]: why do we need additional method if we can gather all txId and further call locks(UUID txId) for each ?

File: modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItLocksSystemViewTest.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        assertThat(objectId, not(emptyString()));
+        assertThat(lockModes, hasItem(mode));
+    }
+
+    private List<Lock> activeLocks(InternalTransaction tx) {
+        List<Lock> res = new ArrayList<>();
+
+        CLUSTER.runningNodes().forEach(node -> unwrapIgniteImpl(node)
+                .txManager()
+                .lockManager()
```
[the remark this answers is not in the document]
Comment (xtern) [reply modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItLocksSystemViewTest.java]: :thinking: reworked test a bit to not use deprecated API.

File: modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java [hunk 3]
```diff
@@ -945,7 +944,7 @@ public class IgniteImpl implements Ignite {
 
         var transactionInflights = new TransactionInflights(placementDriverMgr.placementDriver(), clockService);
 
-        LockManager lockMgr = new HeapLockManager();
+        HeapLockManager lockMgr = new HeapLockManager();
```
[the remark this answers is not in the document]
Comment (xtern) [reply modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java]: Fixed, thanks.

**Review by AMashenkov — APPROVED** [review]

File: modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/HeapLockManager.java (hunk 1 above)
Comment (zstan) [reply modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/HeapLockManager.java:255]: got it, let\`s call also @vldpyatkov here

**Review by zstan — APPROVED** [review]

**Review by vldpyatkov — APPROVED** [review]

**Pull request merged**